### ISSUE TITLE
testInitState fails on OSX, probably due to memory leaks

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#473]
Related to #393.

Also, sometimes testComponents fails as well.

**Comment by chrisdembia** [comment]
```
17: Loaded model PushUpToesOnGround from file PushUpToesOnGroundWithMuscles.osim
17: Model unable to assemble: AssemblySolver::assemble() Failed: SimTK Exception thrown at Assembler.cpp:871:
17:   Method Assembler::assemble() failed because:
17: Optimizer failed with message: SimTK Exception thrown at InteriorPointOptimizer.cpp:261:
17:   Optimizer failed: Ipopt: Restoration failed (status -2)
17: Assembly error tolerance achieved: 5.91402810057318e-09 required: 1e-10.
17: Model relaxing constraints and trying again.
17: Model unable to assemble: AssemblySolver::assemble() Failed: SimTK Exception thrown at Assembler.cpp:871:
17:   Method Assembler::assemble() failed because:
17: Optimizer failed with message: SimTK Exception thrown at InteriorPointOptimizer.cpp:261:
17:   Optimizer failed: Ipopt: Restoration failed (status -2)
17: Assembly error tolerance achieved: 1.36861858651693e-10 required: 1e-10.
17: Model relaxing constraints and trying again.
17: *********************** testMemoryUsage ***********************
17: MODEL: PushUpToesOnGroundWithMuscles.osim uses 11516KB
17: 17872KB change in memory use after 100 state initializations.
17: Approximate leak size: 178.72KB or 1.55193% of model size.
17: Average initialization time: 623.911ms
17: testInitState failed: 
17: Exception:
17:   testMemoryUsage: state initialization leak > 0.5% of model memory
17:   footprint.
17:   file= /Users/chris/repos/opensim/1/opensim-core/OpenSim/Simulation/Test/testInitState.cpp
17:   line= 195
17: 
17: 
1/1 Test #17: testInitState ....................***Failed   63.31 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =  63.31 sec

The following tests FAILED:
         17 - testInitState (Failed)
```

**Comment by sherm1** [comment]
It's odd that the output above says "relaxing constraints and trying again" yet on the second try it still has a required tolerance of 1e-10. And it came **so** close at 1.4e-10! If it had really relaxed the constraint tolerance, say to 1e-9 it would have worked. Thoughts, @aseth1?

**Comment by chrisdembia** [comment]
With #567 Travis now tests on OSX and we can all see the testInitState failure. I've spent some time looking into this and the amount of the leak changes every time. I thought maybe this has to do with unreliability of the system calls being used to detect memory usage, but perhaps that doesn't make sense.

**Comment by chrisdembia** [comment]
Here's some information. On OSX the test usually fails when checking for memory leaks with `initializeState()` for arm26.

- On Ubuntu, the model always consumes 32 kb. If I run the test with Valgrind, the model always consumes 4360 kb, and the test fails (since the initializing the state causes a leak of ~3%). Without valgrind, the test passes.
- On Windows, the model always consumes 584 kb.
- On OSX, the model consumes a variable amount of memory (0 kb, 864 kb, 24 kb, 772 kb, etc.). If I run the test with valgrind, the model's memory consumption becomes repeatable: the model consumes 2536 kb, and the test fails with a ~8.3% leak for initializing the state. Without valgrind, the test sporadically passes.

At first, I thought the issue was with the implementation for `getCurrentRSS()` being unreliable for OSX. That could still be the case. However, I wanted to see if it was OpenSim's fault somehow. So I tried calling `getCurrentRSS()` 4 times in a row. With each execution of the same executable, getCurrentRSS() returned different values, and the subsequent calls sometimes give different results (something like 1036 kb, 1040 kb, 1040kb, 1040 kb is typical). So I started reducing the executable: if the executable only links with simbody or with osimCommon, the output becomes more consistent from execution to execution. When I also link with osimSimulation, the output becomes much more variable (for the same exact executable). I don't really know what's going on but it seems like we can't simply blame the implementation of `getCurrentRSS()`.

This is about as far as my abilities take me...I don't know what to do next, and I don't know how important this is: we can fix the memory leaks by analyzing the other platforms, but I don't know if these issues are the symptom of some larger issue elsewhere.
